File: backend-spring/src/main/java/fr/ubo/kanban/services/TableauService.java
package fr.ubo.kanban.services;

import fr.ubo.kanban.dtos.tableau.TableauRequestDto;
import fr.ubo.kanban.dtos.tableau.TableauResponseDto;

import java.util.List;

public interface TableauService {
    List<TableauResponseDto> findAll();
    TableauResponseDto findById(Long id);
    TableauResponseDto create(TableauRequestDto dto);
    TableauResponseDto update(Long id, TableauRequestDto dto);
    void delete(Long id);
    void ajouterMembre(Long idTableau, Long idUtilisateur);
    void retirerMembre(Long idTableau, Long idUtilisateur);
}

File: frontend/src/services/tableauService.js
import api from './api.js'

export const tableauService = {
  getAll()                        { return api.get('/tableaux') },
  getById(id)                     { return api.get(`/tableaux/${id}`) },
  create(dto)                     { return api.post('/tableaux', dto) },
  update(id, dto)                 { return api.put(`/tableaux/${id}`, dto) },
  delete(id)                      { return api.delete(`/tableaux/${id}`) },
  ajouterMembre(idTableau, idUtilisateur) {
    return api.post(`/tableaux/${idTableau}/membres/${idUtilisateur}`)
  },
  retirerMembre(idTableau, idUtilisateur) {
    return api.delete(`/tableaux/${idTableau}/membres/${idUtilisateur}`)
  },
}